src/query/interpreter.cpp

```diff
@@ -3136,14 +3167,15 @@ Interpreter::PrepareResult Interpreter::Prepare(const std::string &query_string,
         cypher_query = profile_query->cypher_query_;
       }
       if (const auto &clauses = cypher_query->single_query_->clauses_;
-          IsCallBatchedProcedureQuery(clauses) || std::any_of(clauses.begin(), clauses.end(), [](const auto *clause) {
-            return clause->GetTypeInfo() == LoadCsv::kType;
-          })) {
+          IsAllShortestPathsQuery(clauses) || IsCallBatchedProcedureQuery(clauses) ||
+          std::any_of(clauses.begin(), clauses.end(),
+                      [](const auto *clause) { return clause->GetTypeInfo() == LoadCsv::kType; })) {
         // Using PoolResource without MonotonicMemoryResouce for LOAD CSV reduces memory usage.
         // QueryExecution MemoryResource is mostly used for allocations done on Frame and storing `row`s
         query_executions_[query_executions_.size() - 1] = std::make_unique<QueryExecution>(utils::PoolResource(
             128, kExecutionPoolMaxBlockSize, utils::NewDeleteResource(), utils::NewDeleteResource()));
         query_execution_ptr = &query_executions_.back();
+        spdlog::trace("Has all shortest QueryExecutions Resource {}", IsAllShortestPathsQuery(clauses));
```

Comment:
Also refactor this one in similar way to the above one with ternary operator to form a proper sentence